### PR TITLE
Convert ygnmi GoBGP task

### PR DESCRIPTION
**Pull request by wenovus**
Also fixed the following:
* Fixed TODO for op state datastore service. It now behaves the same way as the config datastore.
* Removed the Task interface. The boilerplate savings are substantial.
* Imported latest version of ygot to simply datastore's unmarshalling code.

### CODE REVIEW

File: gnmi/gnmit/gnmit_test.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func toUpd(r *gpb.SubscribeResponse) []*upd {
 // TestONCE tests the subscribe mode of gnmit.
 func TestONCE(t *testing.T) {
 	ctx := context.Background()
-	c, addr, err := New(ctx, nil, "localhost:0", "local", false, nil)
+	c, addr, err := New(ctx, nil, "localhost:0", "local", false)
```
Comment (DanG100) [inline gnmi/gnmit/gnmit_test.go:152]: these tests should stop the server at end, i think that cause the test to fail

**Comment by DanG100** [comment]
thanks for all this refactoring, LGTM pending test fix

File: gnmi/gnmit/gnmit_test.go (hunk 1 above)
Comment (wenovus) [reply gnmi/gnmit/gnmit_test.go:152]: This hit some rare race condition, essentially we can't have listeners listening to it before we try to delete the socket file. I'll take a look at fixing this in a follow-up.

Comment (DanG100) [reply gnmi/gnmit/gnmit_test.go:152]: sgtm

**Review by DanG100 — APPROVED** [review]

**Pull request merged**